### PR TITLE
Introduce custom error pages

### PR DESCRIPTION
**Pull request by sarjon**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | In Back Office when you get an unhandled exception in production you wont be able to see any details regarding that. So its really difficult to provide any support to merchant (its common to receive screenshot from admins with default 500 page which gives you no information) as you dont have any clue of what happened. This PR aims to bring more information about errors that happens in production and possible helps solve them quicker.
| Type?         | new feature
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | n/a
| How to test?  | Try to throw an exception in production environment and you will see an error with additional details regarding the cause. Make sure to clear cache before testing this PR.

Default error page when unhandled exception occurs (may differ depending on your browser/web server):
regular_error_page

New error page when unhandled exception occurs:
51629239-be792a80-1f4f-11e9-96b5-19e573a050f4

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/10780)

### CODE REVIEW

**Review by PierreRambaud — DISMISSED** [review]

File: app/Resources/TwigBundle/views/Exception/error.html.twig
Comment (LouiseBonnard) [inline app/Resources/TwigBundle/views/Exception/error.html.twig]: I would rather localized it in _Admin.Notifications.Error_.

File: app/Resources/TwigBundle/views/Exception/error.html.twig
Comment (LouiseBonnard) [inline app/Resources/TwigBundle/views/Exception/error.html.twig]: Perhaps I would choose a more user-friendly wording here, like _Cheer up, see additional information below for help_. What do you think? @TristanLDD / @colinegin

Comment (sarjon) [reply app/Resources/TwigBundle/views/Exception/error.html.twig]: yes, definetly. :+1: i'd like to update UI as well to match PS styles.

**Comment by colinegin** [comment]
As discussed with Louise, it's a great idea ! Maybe we could also change the first sentence, to make it also more friendly.

File: app/Resources/TwigBundle/views/Exception/error.html.twig
Comment (LouiseBonnard) [reply app/Resources/TwigBundle/views/Exception/error.html.twig]: Check w/ @colinegin, it could be cool to have a more user-friendly tone at this point, in order to match the following empathetic wording: _Oops... looks like an unexpected error occurred_.

**Comment by matks** [comment]
@colinegin So what do we do ? Do we work on this with @TristanLDD and @LouiseBonnard ?

**Comment by sarjon** [comment]
it would be nice to get a bit better looking mockup for this. :)

**Comment by MathiasReker** [comment]
How about some similar to this? :)

**Comment by sarjon** [comment]
> How about some similar to this? :)

yes, except we could use Preston. :)

**Comment by MathiasReker** [comment]
Yes, using Preston would be ideal! 👍

**Comment by colinegin** [comment]
@TristanLDD what do you think ?

**Comment by TristanLDD** [comment]
Indeed, using a friendly ton of voice + illustration on this page would be ideal. But with the new branding coming, using Preston may not be the best solution. I need to work on the design, I will upload the mockup here asap.

**Comment by mickaelandrieu** [comment]
@TristanLDD do we have some error page for 404 also? I think it's the right time to do both not found/404 and error/500 pages at the same time if you plan to work on it 👍 

Ping @sarjon, to be confirmed but I'm pretty sure these pages can be overridden in PrestaShop modules, too :)

**Comment by sarjon** [comment]
@mickaelandrieu im thinking about moving actual template to bundle, like so:

```twig
{# app/Resources/TwigBundle/views/Exception/error.html.twig #}

{% include '@Prestashop/bundle/path/to/error/template.html.twig' %}
```
 so all of our templates live in the same place. :) and yeah, it should be able to override them from modules as well.

**Comment by TristanLDD** [comment]
@mickaelandrieu Yes I'll do both 404 and 500 ;)

**Comment by NadiaCamard** [comment]
Hello,
For this topic Tristan hands over to me, I will work on the design for 404 and 500.
But first I need more info about technical restrictions:

- should the design be contained in a specific width?
- Is the PrestaShop header (with all the links) necessary?
- On 404 page, where should it redirect? (Previous page? Dashboard?)
- Are these designs only required for the Back Office? In which case we will display these pages?

Thanks in advance for your reply :)

**Comment by sarjon** [comment]
> Are these designs only required for the Back Office? In which case we will display these pages?

yes, only for Back Office. we will display 404 pages in case user tries to access page that does not exist and 500 page in case fatal error occurs. :)

other questions i cant answer, maybe @colinegin can. :)

**Comment by NadiaCamard** [comment]
Hello,
Here are the mockups for 404 and 500.
error 404
error 500
https://invis.io/DTOJVV3YM34
For visual specifications, please ask me :)

**Comment by sarjon** [comment]
@NadiaCamard looks really nice!

how do i download images?

**Comment by NadiaCamard** [comment]
@sarjon Hello, I just sent you an email. Thanks :)

**Comment by sarjon** [comment]
quick notices regarding 500 page

1. i think that we should render only card (without top and side menus) because rendering whole page when there is an unhandled exception is risky. For example, image there is an issue for rendering side menu that throws exception, this page would never load.

2. I dont think that suggestion to enable Debug mode on production is a good idea unless we implement this feature => https://github.com/PrestaShop/PrestaShop/issues/11038

@matks wdyt?

**Comment by matks** [comment]
@sarjon 
> 1. i think that we should render only card (without top and side menus)

For 404, we should be fine rendering the menu. This is useful to allow back-office users to "get back to business" using menu links, especially if he clicked on an old link and needs the new one.

For 500 ... you comment makes sense. But once again, having the menu is a nice-to-have for the UX (correct me if I'm wrong 😅 this is not my field). I guess the best would be to *try* to render the menu but if it fails, it would not prevent the error from being displayed. I have no idea how hard this would be to implement though.

> 2. I dont think that suggestion to enable Debug mode on production is a good idea unless we implement this feature

Merchants can already enable it on production, and if they do not, people advise it to them on the community forum. Are you worried that merchants would turn on the debug mode while being unaware of the consequences ? If yes, my opinion is that we should *educate the user* about the debug mode and what it implies, rather than "not talking about it", don't you think 😉 ? We could do that with Helper Cards, red warning messages ...

https://github.com/PrestaShop/PrestaShop/issues/11038 is still a very good idea 😋

**Comment by sarjon** [comment]
i've noticed one issue regarding 404 page. If NotFoundHttpException is thrown, the request is fowarded to legacy dispatcher, so we wont be able to see our nice page. ;(

For devs, see code https://github.com/PrestaShop/PrestaShop/blob/develop/admin-dev/index.php#L91

**Comment by mickaelandrieu** [comment]
> i've noticed one issue regarding 404 page. If NotFoundHttpException is thrown, the request is fowarded to legacy dispatcher, so we wont be able to see our nice page. ;(

Noted, I think this can be part of another contribution.

**Review by mickaelandrieu — DISMISSED** [review]

**Comment by marionf** [comment]
Hello @sarjon 

For 500 error, buttons "back to previous page" and "enable debug mode" are not working

For 404 error, I still have this:

capture d ecran_748

**Comment by sarjon** [comment]
@marionf when user clicks "Enable debug mode", should it redirect to Performance page (keep in mind it could be broken since user got 500 page) or should we enable Debug Mode right away? If so, what should happen after enabling Debug Mode? Should we redirect user back to original page where 500 occured?

**Comment by marionf** [comment]
@sarjon I think it's better to enable debug mode right away and redirect user to original page, this way the user has more details about the error.
But let's wait for the confirmation of @NadiaCamard and @colinegin

**Comment by sarjon** [comment]
@matks could you review? lets finish this PR! :checkered_flag: 

@LouiseBonnard can you check wording? some keys dont even have domains, so im looking foward to your suggestions. ^^

@TristanLDD this is the furthest i can go with styling, see below. 

What i dont like is that "Error code", "Error type" and "Error message" are centered, maybe it would be better to show them as single block of information since they are related? something like table or left align so its easier to read, wdyt?

File: src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig
Comment (sarjon) [inline src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig]: image does not go through webpack build process, thus it is not in `/public`, should we change that or keep as is?

File: src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig]: Please localize it in _Admin.Notifications.Error_, thanks!

File: src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig]: Same here...

File: src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig]: ... and here also! :-D

Comment (LouiseBonnard) [reply src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig]: Is this message necessary? I mean, wouldn't it be possible to only write "Enable debug mode and try again to track the issue." without adding "Error message:" before?

File: src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig]: _Back to the previous page_ already exists in the back office, perhaps it is better to use it then? Although I prefer _Back to previous page_, without the article. And I think it should be localize it in _Admin.Actions_.

File: src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig]: Same, I would localize it in _Admin.Actions_. :-)

File: src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig
Comment (LouiseBonnard) [reply src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig]: And I agree with @sarjon, the fact it is centered this way is kind of upseting me... is it a design bias?

File: src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig]: In _Admin.Notifications.Error_ as well, thank you.

**Comment by NadiaCamard** [comment]
@sarjon Hello!
Yes for text styling, we can left align that.
After discussion, it would be better to refine the priority level of each call-to-action: "Back to previous" button should be more important and "Enable debug mode" should be an outlined button, set on the left.
Also it would good to add a link to a tutorial. Many users are not familiar with the debug mode.
Thanks again!
error 500

File: src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig
Comment (matks) [reply src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig]: Do you mean than if you put it in `/public`, there is a webpack error ?

File: src/Core/Domain/Configuration/Command/EnableDebugModeCommand.php
Comment (matks) [inline src/Core/Domain/Configuration/Command/EnableDebugModeCommand.php]: I'm considering the idea of turning this `EnableDebugModeCommand` into `SwicthDebugModeCommand` which would be able, using a boolean input, to enable or disable the debug mode. Wdyt @sarjon ?

**Comment by sarjon** [comment]
@NadiaCamard where should "Learn more about debug mode" redirect?

File: src/Core/Domain/Configuration/Command/EnableDebugModeCommand.php
Comment (sarjon) [reply src/Core/Domain/Configuration/Command/EnableDebugModeCommand.php]: hmmm, i guess it could be. what values it should accept?

**Comment by NadiaCamard** [comment]
The "Learn more..." link should redirect to:
EN : http://doc.prestashop.com/display/PS17/Performance#Performance-Debugmode
FR : http://doc.prestashop.com/pages/viewpage.action?pageId=51840325#Param%C3%A8tresdeperformances-Modedebug
IT : http://doc.prestashop.com/pages/viewpage.action?pageId=54266514
ES : http://doc.prestashop.com/display/PS17/Rendimiento#Rendimiento-ModoDepuraci%C3%B3n
other languages : http://doc.prestashop.com/display/PS17/Performance#Performance-Debugmode

File: src/Core/Domain/Configuration/Command/EnableDebugModeCommand.php
Comment (matks) [reply src/Core/Domain/Configuration/Command/EnableDebugModeCommand.php]: Just a boolean, set to "true" to enable and to "false" to disable. It does the job, like https://github.com/PrestaShop/PrestaShop/pull/11894/files#diff-b3ae8c4352f26f962b5fb157163d6f6e

**Comment by colinegin** [comment]
Any news about this page @matks and @sarjon ?

**Comment by sarjon** [comment]
still WIP, hope to finish it this week. :)

**Comment by sarjon** [comment]
this is getting closer! :heart_eyes:

The one thing i still dont like is text. :/ when it's centered, it looks bad, when it's aligned left (tried different positions) still looks bad...

dbm

however, what i suggest is: 

1. Cut off these "Error message:", "Error type:" and "Error code:" texts.
2. Keep only error message centered.
3. On second line output exception and it's code.

imo, it looks much cleaner and is easy to read, see example below.

dbas

@colinegin @NadiaCamard wdyt?

File: src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig
Comment (sarjon) [inline src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig:73]: @matks wdyt about this `documentation_link` function?

File: src/PrestaShopBundle/Resources/config/services/bundle/twig.yml
Comment (sarjon) [inline src/PrestaShopBundle/Resources/config/services/bundle/twig.yml]: @matks i think that this service should be moved under `Core/Documentation` namespace since it can be used not only for cards, wdyt?

**Comment by NadiaCamard** [comment]
@sarjon Thanks! Yes it looks better!
Your suggestions are appropriate: Keep the text centered and reduce it.

Concerning the line output exception and it's code, can you shrink it? It seems common to show only the error code.

Also can you add more space between the 2 CTAs? (20 pixels)

**Comment by sarjon** [comment]
> Concerning the line output exception and it's code, can you shrink it? It seems common to show only the error code.

the exception name is a must, since codes can be the same across multiple exceptions.

**Comment by NadiaCamard** [comment]
@sarjon Ok let's leave this line the way it is :)

File: src/PrestaShopBundle/Resources/config/services/bundle/twig.yml
Comment (matks) [reply src/PrestaShopBundle/Resources/config/services/bundle/twig.yml]: Yes we can 👍 we just need to a service alias to preserve backwards-compatibility

File: src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig
Comment (matks) [reply src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig]: Did you want to put it through webpack processing, but could not because of an issue, or did you want to avoid the compilation step for this process and ask if it's alright ?

File: src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig
Comment (matks) [reply src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig:73]: It's fine 👍 but we need to reuse it in order to make it worth

**Comment by sarjon** [comment]
waiting for QA? :)

**Comment by marionf** [comment]
@sarjon 

For 404 I still have this

capture d ecran_974

For 500 it's all good :+1: 

capture d ecran_975

**Comment by marionf** [comment]
Also in RTL language, it's not reversed

capture d ecran_976

**Comment by sarjon** [comment]
@marionf 404 page will not be implemented because of legacy support. :/

**Comment by colinegin** [comment]
@sarjon what do you mean exactly for the 404 ? 
This can happen both on legacy and symfony pages, just like error 500, right ?

**Comment by sarjon** [comment]
@colinegin Symfony allows handling these kind of errors (500, 404 & etc) with custom pages. This is what I did with 500 error.

However, if 404 occurs in migrated pages it fowards request to legacy part, so it can be handled by legacy controller. Here's my previous, more technical, comment regarding this:

> i've noticed one issue regarding 404 page. If NotFoundHttpException is thrown, the request is fowarded to legacy dispatcher, so we wont be able to see our nice page.

even though i think 404 could somehow be implemented somehow, it should be part of another PR.

**Comment by colinegin** [comment]
I understand @sarjon , thanks for your comment !

**Comment by NadiaCamard** [comment]
@sarjon For RTL version can you:
- reverse the illustration
- reverse the order of the CTAs
- reverse the reading direction of the arrow icon contained in the link "Learn more..."
?

**Comment by colinegin** [comment]
@matks any news about the rtl behaviour ?

**Comment by sarjon** [comment]
@matks can we merge and fix RTL during freeze (create issue for that)?

**Comment by sarjon** [comment]
ping @jolelievre this is not migration related PR, but it would be nice to merge before freeze so we can have this feature in 1.7.6. :tada:

File: admin-dev/themes/new-theme/public/sql_manager.bundle.js
Comment (jolelievre) [inline admin-dev/themes/new-theme/public/sql_manager.bundle.js]: your assets are in dev prod, some contain conflicts, and some are in conflict with develop branch anyway ^^
I think you should rebase and regenerate them

File: src/PrestaShopBundle/Controller/Admin/ErrorController.php
Comment (jolelievre) [inline src/PrestaShopBundle/Controller/Admin/ErrorController.php:53]: Is this always present? set by Symfony framework?

File: src/PrestaShopBundle/Resources/config/routing/admin/_errors.yml
Comment (jolelievre) [inline src/PrestaShopBundle/Resources/config/routing/admin/_errors.yml:1]: This should be `POST` only no?

File: src/PrestaShopBundle/Controller/Admin/ErrorController.php
Comment (jolelievre) [inline src/PrestaShopBundle/Controller/Admin/ErrorController.php:48]: I think an `@AdminSecurity` is mandatory here

File: src/PrestaShopBundle/Resources/config/services/bundle/twig.yml
Comment (jolelievre) [reply src/PrestaShopBundle/Resources/config/services/bundle/twig.yml]: Agreed!

File: src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig
Comment (jolelievre) [inline src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig:60]: No csrf token protection?

**Review by jolelievre — DISMISSED** [review]

**Comment by jolelievre** [comment]
@sarjon a few things to improve, but yes it would be great to have this
Let's be careful about what is displayed however, we had security fixes on 1.6 versions because the error page displayed informations like the database credentials.. But with just the exception message (no stacktrace) I think we should be fine

File: src/PrestaShopBundle/Controller/Admin/ErrorController.php
Comment (sarjon) [reply src/PrestaShopBundle/Controller/Admin/ErrorController.php:53]: No, developer has to set it.

File: src/PrestaShopBundle/Resources/config/routing/admin/_errors.yml
Comment (sarjon) [reply src/PrestaShopBundle/Resources/config/routing/admin/_errors.yml:1]: I think you're right. :+1:

File: src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig
Comment (sarjon) [reply src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig:60]: Yeah... :/ But I think URL itself has `token` and if that token is invalid, you won't be able to submit it.

File: src/PrestaShopBundle/Twig/Extension/DocumentationLinkExtension.php
Comment (sarjon) [inline src/PrestaShopBundle/Twig/Extension/DocumentationLinkExtension.php]: should we make extensions `final`?

File: src/PrestaShopBundle/Controller/Admin/ErrorController.php
Comment (sarjon) [reply src/PrestaShopBundle/Controller/Admin/ErrorController.php:48]: Hmmm, this is interesting case. `@AdminSecurity` works by using legacy controllers, but this is new controller that does not have equivalent controller in legacy.

File: src/PrestaShopBundle/Controller/Admin/ErrorController.php
Comment (jolelievre) [reply src/PrestaShopBundle/Controller/Admin/ErrorController.php:53]: Then maybe we should have a fallback in case it is not set

Comment (jolelievre) [reply src/PrestaShopBundle/Controller/Admin/ErrorController.php:53]: Oh ok, it comes from the form! Good!

File: src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig
Comment (jolelievre) [reply src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig:60]: That's what I figured, but I was not sure.

File: src/PrestaShopBundle/Twig/Extension/DocumentationLinkExtension.php
Comment (jolelievre) [reply src/PrestaShopBundle/Twig/Extension/DocumentationLinkExtension.php]: Maybe we should, I think using `final` on internal classes "protects" us in a way if we need to change their constructor or signature, so that it is not (less?) considered as a BC break

File: src/Core/Domain/Configuration/CommandHandler/SwitchDebugModelHandlerInterface.php
Comment (rokaszygmantas) [inline src/Core/Domain/Configuration/CommandHandler/SwitchDebugModelHandlerInterface.php]:
```suggestion
interface SwitchDebugModeHandlerInterface
```

**Comment by matks** [comment]
Once this is merged, we need to remove the try/catch used in places like [here](https://github.com/PrestaShop/PrestaShop/blob/develop/src/Adapter/Currency/CommandHandler/AddCurrencyHandler.php#L83) as they will be handled better by this PR

File: src/Adapter/Configuration/CommandHandler/SwitchDebugModeHandler.php
Comment (matks) [inline src/Adapter/Configuration/CommandHandler/SwitchDebugModeHandler.php]: Why do you check `$this->debugMode->isDebugModeEnabled())` for the "enable debug mode" usecase but for the "disable debug mode" you check nothing ?

File: src/Core/Domain/Configuration/CommandHandler/SwitchDebugModelHandlerInterface.php
Comment (matks) [reply src/Core/Domain/Configuration/CommandHandler/SwitchDebugModelHandlerInterface.php]: 🆙

File: src/PrestaShopBundle/Controller/Admin/ErrorController.php
Comment (matks) [reply src/PrestaShopBundle/Controller/Admin/ErrorController.php:48]: Maybe we could require the same access rights than the ones for the Maintenance page where people can enable/disable debug mode ? I think it would make sense

For the record, if we create new pages, it's not an issue not to have a legacy controller. We can create fake entries in the tab SQL table to be used for the new page.

File: src/PrestaShopBundle/Resources/config/routing/admin/_errors.yml
Comment (matks) [reply src/PrestaShopBundle/Resources/config/routing/admin/_errors.yml:1]: 🆙

File: src/PrestaShopBundle/Resources/config/services/bundle/twig.yml
Comment (matks) [reply src/PrestaShopBundle/Resources/config/services/bundle/twig.yml]: 🆙

File: src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig
Comment (matks) [reply src/PrestaShopBundle/Resources/views/Admin/Exception/error.html.twig]: 🆙

**Comment by matks** [comment]
@sarjon Finally reviewed 😅 sorry for the wait

I've put a few minor questions and a few 🆙for unhandled requested changes

We can handle RTL in a 2nd PR

File: src/PrestaShopBundle/Controller/Admin/ErrorController.php
Comment (sarjon) [reply src/PrestaShopBundle/Controller/Admin/ErrorController.php:48]: > We can create fake entries in the tab SQL table to be used for the new page.

Yeah, but then we also need update "Permissions" page, right? I'll use "Performance" page as it is the place where debug mode can be configured.

File: src/PrestaShopBundle/Resources/config/routing/admin/_errors.yml
Comment (sarjon) [reply src/PrestaShopBundle/Resources/config/routing/admin/_errors.yml:1]: Done.

**Comment by sarjon** [comment]
Addressed comments.

File: src/PrestaShopBundle/Controller/Admin/ErrorController.php
Comment (matks) [inline src/PrestaShopBundle/Controller/Admin/ErrorController.php]:
```suggestion
     * @AdminSecurity("is_granted(['update', 'create', 'delete'], 'AdminPerformance')")
```
Directly ?

Comment (sarjon) [reply src/PrestaShopBundle/Controller/Admin/ErrorController.php]: Yeah, that would work, done. :+1:

**Review by matks — DISMISSED** [review]

**Comment by sarjon** [comment]
QA ready?

**Comment by matks** [comment]
@sarjon Yes 😄 
@LouiseBonnard One last review for the "Wording ✔️ " label 😄 ?

**Comment by marionf** [comment]
It's all good for error 500

capture d'écran_1711

It remains the RTL to manage

**Comment by matks** [comment]
> It remains the RTL to manage

I'm afraid handling the RTL would require a lot of extra work. Unfortunately this is an error page: it appears when something went wrong in the system. That is why we cannot rely on the system to behave correctly, so I dont think we can retrieve the "is the language used a RTL language" information in a manner that is both accurate and reliable.

I started working on it and after a few obstacles, I see it now become a real problem.

So I suggest we dont handle the RTL. I dont think a merchant will be angry at us if the error page image is not RTL 😄 keep in mind that, in a nominal case, he will never see it 😉

**Comment by matks** [comment]
I need to rebase before merge

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
Waiting for CI ✅to merge

**Comment by sarjon** [comment]
> Waiting for CI ✅to merge

CI said no. 😢

**Comment by matks** [comment]
> > Waiting for CI ✅to merge
> 
> CI said no. 😢

I had a talk with Travis and now we're good

**Pull request merged**
